src/Avalonia.Styling/Styling/PropertySetterBindingInstance.cs

```diff
@@ -90,9 +93,12 @@ public void Start(bool hasActivator)
 
         public void Activate()
         {
+            if (_binding is null)
+                return;
+
             if (!_isActive)
             {
-                _innerSubscription ??= _binding.Observable.Subscribe(_inner);
+                _innerSubscription ??= _binding.Observable!.Subscribe(_inner);
```

Comment:
So, technically null binding instance was supported, but never worked in fact? Otherwise there would be NRE.

Reply:
Yeah `InstancedBinding` is just a terrible API (I'm sorry). If I've read the code correctly, `InstancedBinding.Observable` will only be null for a `OneTime` binding which we explicitly disallow in the constructor. This can be seen in the `InstancedBinding` static factory methods: they all pass an `IObservable` as the value except `OneTime`.

Reply:
Oh wait, maybe you're talking about something else. Yeah previously we would have got an NRE for a null `InstancedBinding`, now we just ignore it. But maybe it would be better to throw an exception in the `PropertySetterBindingInstance` constructor if `binding.Initiate` returns null.